packages/flutter/lib/src/services/text_boundary.dart

```diff
@@ -123,9 +123,92 @@ class LineBoundary extends TextBoundary {
   TextRange getTextBoundaryAt(int position) => _textLayout.getLineAtOffset(TextPosition(offset: max(position, 0)));
 }
 
+/// A text boundary that uses paragraphs as logical boundaries.
+///
+/// A paragraph is defined as the range between line terminators. If no
+/// line terminators exist then the paragraph boundary is the entire document.
+class ParagraphBoundary extends TextBoundary {
+  /// Creates a [ParagraphBoundary] with the text.
+  const ParagraphBoundary(this._text);
+
+  final String _text;
+
+  /// Returns the [int] representing the start position of the paragraph that
```

Comment:
Perhaps remove the doc comment reference to [int] as it's a generic type? Can follow the style of `TextBoundary` (or even `CharacterBoundary` which only has class level doc comments...).